src/core/Akka/Configuration/Hocon/HoconParser.cs

```diff
@@ -73,11 +76,25 @@ private void ParseObject(HoconValue owner, bool root)
                 Token t = _reader.PullNext();
                 switch (t.Type)
                 {
+                    case TokenType.Include:
+                        var included = _includeCallback(t.Value);
+                        var substitutions = included.Substitutions;
+                        foreach (var substitution in substitutions)
+                        {
+                            //fixup the substitution, add the current path as a prefix to the substitution path
+                            substitution.Path = currentPath + "." + substitution.Path;
+                        }
+                        _substitutions.AddRange(substitutions);
+                        var otherObj = included.Value.GetObject();
+                        owner.GetObject().Merge(otherObj);
+
```

Comment:
if the token is an Include, then we call the `_includeCallback` to get the requested sub object.
the sub object is then merged with the currently parsed hocon object.
any substitutions in the sub object is copied into the current parsing pass, and each of those substitutions are rewritten to point to the sub objects new location in the object hierarchy.